File: src/main/java/br/com/kronos/backend/aplicacao/pessoa/Talento.java
package br.com.kronos.backend.aplicacao.pessoa;

import static java.util.Objects.requireNonNull;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Talento {

	private int id;
	private Long idPessoa;
	private int idTipoTalento;
	
	@Builder
	public Talento(int id, Long idPessoa, int idTipoTalento) {
		this.id = id;
		this.idPessoa = requireNonNull(idPessoa, "idPessoa é obrigatória");
		this.idTipoTalento = requireNonNull(idTipoTalento, "idTipoTalento é obrigatório");
	}
	
}

File: poc-kronos-frontend/src/model/pessoa/Talento.js
class Talento {
  constructor(dados) {
    this.idPessoa = dados ? dados.idPessoa : null;
    this.idTipoTalento = dados ? dados.idTipoTalento : null;
    this.idsTiposTalentos = null;
    this.nomeTipoTalento = dados ? dados.nomeTipoTalento : null;
  }

  static validations(required) {
    return {
      idPessoa: { required },
      idsTiposTalentos: { required }
    };
  }

  static validationMessages() {
    return {
      idPessoa: {
        required: "Pessoa é obrigatória"
      },
      idsTiposTalentos: {
        required: "Tipo Talento é obrigatório"
      }
    };
  }
}
export default Talento;
